cmd/dp-recipe-api/main.go

```diff
@@ -41,18 +43,19 @@ func main() {
 	//Feature Flag for Mongo Connection
 	enableMongoData := cfg.MongoConfig.EnableMongoData
 
-	if enableMongoData {
-		mongodb := &mongo.Mongo{
-			Collection: cfg.MongoConfig.Collection,
-			Database:   cfg.MongoConfig.Database,
-			URI:        cfg.MongoConfig.BindAddr,
-		}
+	mongodb := &mongo.Mongo{
+		Collection: cfg.MongoConfig.Collection,
+		Database:   cfg.MongoConfig.Database,
+		URI:        cfg.MongoConfig.BindAddr,
+	}
 
-		var err error
-		mongodb.Session, err = mongodb.Init()
+	if enableMongoData {
+		session, err := mongodb.Init()
```

Comment:
I think you can still assign directly to `mongodb.Session` here:
```
mongodb.Session, err = mongodb.Init()
``` 
this will remove the need for the else statement?

Reply:
If I initialise `err` before that, then I can do the above. Therefore, I have done that and committed in "Simplifying code and improving gracefulShutdown."